src/toast/tests/ops_crosstalk.py

```diff
@@ -0,0 +1,214 @@
+# py37+
+# from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import numpy as np
+from numpy.random import default_rng
+
+from ..dist import Data
+from ..mpi import get_world
+from ..tod import TODCache
+from ..tod.crosstalk import OpCrosstalk, SimpleCrosstalkMatrix
+from ..utils import Logger
+from ._helpers import create_outdir  # , create_satellite_data
+from .mpi import MPITestCase
+
+if TYPE_CHECKING:
+    from typing import List
+
+
+mpiworld, procs, rank = get_world()
+log = Logger.get()
+
+# just to have more than 1 detectors per procs, and is irregular
+n_detectors = 4 * procs + 3
+n_samples = 100
+signal_name = "signal"
+detrankses = [1]
+for i in range(2, procs + 1):
+    if procs % i == 0:
+        detrankses.append(i)
+log.info(f'Testing OpCrosstalk against detranks {", ".join(map(str, detrankses))}.')
+
+names_str = [f"A{i}" for i in range(n_detectors)]
+names = np.array(names_str, dtype="S")
+names_2_str = [f"B{i}" for i in range(n_detectors)]
+names_2 = np.array(names_str, dtype="S")
+
+tod_array = np.arange(n_detectors * n_samples, dtype=np.float64).reshape(
+    (n_detectors, n_samples)
+)
+crosstalk_data = np.arange(n_detectors * n_detectors, dtype=np.float64).reshape(
+    (n_detectors, n_detectors)
+)
+tod_crosstalked = (
+    np.arange(n_detectors * n_detectors).reshape((n_detectors, n_detectors))
+    @ np.arange(n_detectors * n_samples).reshape((n_detectors, n_samples))
+).astype(np.float64)
+
+rng = default_rng()
+tod_array_random = rng.standard_normal((n_detectors, n_samples))
+crosstalk_data_2 = np.identity(n_detectors) + np.reciprocal(
+    np.arange(10, 10 + n_detectors * n_detectors)
+).reshape((n_detectors, n_detectors))
+tod_crosstalked_random = crosstalk_data_2 @ tod_array_random
```

Comment:
All of this module level code should be moved into the `MPITestCase` class below.  The `setUp(self)` method has access to the global communicator in `self.comm`.  So you could initialize your data in that method.